app/src/main/java/com/pvp/app/ui/screen/calendar/TaskFormsCommon.kt

```diff
@@ -1,190 +1,278 @@
+    var description by remember {
+        mutableStateOf(
+            (task as? MealTask)?.recipe ?: task?.description ?: ""
+        )
+    }
+    var activity by remember {
+        mutableStateOf(
+            (task as? SportTask)?.activity ?: SportActivity.Walking
+        )
+    }
+    var duration by remember { mutableStateOf(task?.duration) }
+    var distance by remember { mutableStateOf((task as? SportTask)?.distance) }
+    var reminderTime by remember { mutableStateOf(task?.reminderTime) }
+    var dateTime by remember {
+        mutableStateOf(
+            if (task?.date != null && task.time != null) {
+                LocalDateTime.of(task.date, task.time)
+            } else date ?: LocalDateTime.now()
+        )
+    }
+
+    val isFormValid by remember {
+        derivedStateOf {
+            title.isNotEmpty()
+        }
+    }
+
+    Column(
+        modifier = Modifier
+            .background(MaterialTheme.colorScheme.surfaceContainer)
+            .clip(MaterialTheme.shapes.medium)
+            .padding(8.dp)
+            .fillMaxWidth(),
+        verticalArrangement = Arrangement.spacedBy(16.dp)
+    ) {
+        EditableTextItem(
+            label = "Title",
+            value = title,
+            onValueChange = { title = it }
+        )
+
+        EditableTextItem(
+            label = if (taskClass == MealTask::class) "Recipe" else "Description",
+            value = description,
+            onValueChange = { description = it }
+        )
+
+        if (targetClass == SportTask::class) {
+            EditableSportActivityItem(
+                label = "Activity",
+                value = activity,
+                onValueChange = { activity = it }
+            )
+        }
+
+        if (taskClass == SportTask::class && activity.supportsDistanceMetrics) {
+            EditableDistanceItem(
+                label = "Distance",
+                value = distance,
+                rangeKilometers = model.rangeKilometers,
+                rangeMeters = model.rangeMeters,
+                onValueChange = { distance = it }
+            )
         }
-    )
 
-    if (activity != null && activity!!.supportsDistanceMetrics) {
-        val stateKilometers = PickerState.rememberPickerState(
-            task?.distance
-                ?.toInt()
-                ?: model.rangeKilometers.first()
+        if (taskClass != SportTask::class || !activity.supportsDistanceMetrics) {
+            EditablePickerItem(
+                label = "Duration",
+                value = duration,
+                valueLabel = "minutes",
+                items = model.rangeDuration,
+                itemsLabels = "minutes",
+                onValueChange = { duration = it }
+            )
+        }
+
+        EditableTimeItem(
+            label = "Scheduled at",
+            value = dateTime.toLocalTime(),
+            valueDisplay = if ((taskClass == SportTask::class && activity.supportsDistanceMetrics)
```

Comment:
Split after every logical operator `&&`, `||`. Each condition should be in its own line. First condition should also be in its own line, not behind `if (`